cfme/fixtures/testcases_tools.py

```diff
@@ -0,0 +1,319 @@
+    )
+    group.addoption(
+        "--jsons-no-blacklist",
+        action="store_true",
+        default=False,
+        help="don't filter testcases using the built-in blacklist",
+    )
+
+
+def get_unicode_str(obj):
+    """Makes sure obj is a unicode string."""
+    try:
+        # Python 2.x
+        if isinstance(obj, unicode):
+            return obj
+        if isinstance(obj, str):
+            return obj.decode("utf-8", errors="ignore")
+        return unicode(obj)
+    except NameError:
+        # Python 3.x
+        if isinstance(obj, str):
+            return obj
+        if isinstance(obj, bytes):
+            return obj.decode("utf-8", errors="ignore")
+        return str(obj)
+
+
+def _get_docstring(item):
+    try:
+        docstring = get_unicode_str(item.function.func_doc)
+    except Exception:
+        docstring = ""
+    return docstring
+
+
+def _get_caselevel(item):
+    try:
+        tier = item.get_marker("tier").args[0]
+        tier_id = CASELEVELS[str(tier)]
+    except Exception:
+        tier_id = CASELEVELS["0"]
+    return tier_id
+
+
+def _get_description(item, docstring, automation_script):
+    try:
+        description = docparser.strip_polarion_data(docstring)
+    except ValueError as err:
+        print("Cannot parse the description of {}: {}".format(item.location[2], err))
+        description = ""
+
+    if not item.get_marker("manual"):
+        # Description with timestamp and link to test case source.
+        # The timestamp will not be visible in Polarion, but will cause Polarion
+        # to update the "Updated" field even when there's no other change.
+        description = '{0}<br id="{1}"/><br/><a href="{2}">Test Source</a>'.format(
+            description, TIMESTAMP, automation_script
+        )
+
+    return description
+
+
+def _get_automation_script(item):
+    if item.get_marker("manual"):
+        return None
+
+    # The master here should probably link the latest "commit" eventually
+    automation_script = "http://github.com/{0}/{1}/blob/master/{2}#L{3}".format(
+        CONF["gh_owner"], CONF["gh_repo"], item.location[0], item.function.func_code.co_firstlineno
+    )
+
+    return automation_script
+
+
+def _get_steps_and_results(parsed_docstring):
+    if not parsed_docstring.get("testSteps"):
+        return None
+
+    test_steps = []
+    expected_results = []
+
+    steps = parsed_docstring.get("testSteps")
+    results = parsed_docstring.get("expectedResults")
+    steps = [STEP_NUMBERING.sub("", s[2]) for s in steps]
+    results = [STEP_NUMBERING.sub("", r[2]) for r in results]
+
+    for index, step in enumerate(steps):
+        test_steps.append(step)
+
+        try:
+            result = results[index]
+        except IndexError:
+            result = ""
+        expected_results.append(result)
+
+    return test_steps, expected_results
+
+
+def _get_assignee(parsed_docstring):
+    assignee = parsed_docstring.get("assignee")
+    if assignee and assignee[2]:
```

Comment:
Actually it is a tuple with ``lineno, column, value`` (because the library is also used for flake8 as a linter, it returns the position of the record as well). This way it tests that the record is there, the value is not empty and then returns the value.